File: pythonProject/sobrecargaMetodo.py
class Calculadora:
    def somar(self, a, b, c=None):
        if c is None:
            return a + b
        else:
            return a + b + c

# Testando a Calculadora
calc = Calculadora()
print(calc.somar(2, 3))  # Dois parâmetros
print(calc.somar(2, 3, 4))  # Três parâmetros

File: ProjetoJava/src/sobrecargaMetodo.java
class Calculadora {
    public int somar(int a, int b) {
        return a + b;
    }

    public int somar(int a, int b, int c) {
        return a + b + c;
    }

    public static void main(String[] args) {
        Calculadora calc = new Calculadora();
        System.out.println(calc.somar(2, 3));  // Dois parâmetros
        System.out.println(calc.somar(2, 3, 4));  // Três parâmetros
    }
}
